sockeye/layers.py

```diff
@@ -46,11 +48,1542 @@ def activation(data: mx.sym.Symbol, act_type: str) -> mx.sym.Symbol:
+
+        :param x: Symbol of shape (batch_size, seq_len, num_hidden)
+        :return: Symbol of shape (batch_size, seq_len, num_hidden)
+        """
+        data = mx.sym.Activation(data, act_type=self.act_type)
+        return data
+
+    def get_num_hidden(self) -> int:
+        return self.num_hidden
+
+
+class ActivationLayerConfig(LayerConfig):
+
+    def __init__(self, act_type: str = C.RELU):
+        super().__init__()
+        self.act_type = act_type
+
+    def create_encoder_layer(self, input_num_hidden: int, prefix: str) -> EncoderLayer:
+        return StatelessBlockLayer(ActivationBlock(num_hidden=input_num_hidden, act_type=self.act_type))
+
+    def create_decoder_layer(self, input_num_hidden: int, prefix: str) -> DecoderLayer:
+        return StatelessBlockLayer(ActivationBlock(num_hidden=input_num_hidden, act_type=self.act_type))
+
+
+class DropoutBlock(StatelessBlock):
+    """
+    Position-wise feed-forward network with activation.
+    """
+
+    def __init__(self,
+                 dropout: float,
+                 num_hidden: int) -> None:
+        self.dropout = dropout
+        self.num_hidden = num_hidden
+
+    def __call__(self,
+                 data: mx.sym.Symbol,
+                 lengths: Optional[mx.sym.Symbol] = None,
+                 max_length: Optional[int] = None) -> mx.sym.Symbol:
+        """
+        Position-wise feed-forward network with activation.
+
+        :param x: Symbol of shape (batch_size, seq_len, num_hidden)
+        :return: Symbol of shape (batch_size, seq_len, num_hidden)
+        """
+        if self.dropout > 0.0:
+            data = mx.sym.Dropout(data, p=self.dropout)
+        return data
+
+    def get_num_hidden(self) -> int:
+        return self.num_hidden
+
+
+class DropoutLayerConfig(LayerConfig):
+
+    def __init__(self, dropout: float):
+        super().__init__()
+        self.dropout = dropout
+
+    def create_encoder_layer(self, input_num_hidden: int, prefix: str) -> EncoderLayer:
+        return StatelessBlockLayer(DropoutBlock(num_hidden=input_num_hidden, dropout=self.dropout))
+
+    def create_decoder_layer(self, input_num_hidden: int, prefix: str) -> DecoderLayer:
+        return StatelessBlockLayer(DropoutBlock(num_hidden=input_num_hidden, dropout=self.dropout))
+
+
+class LearnedAdditivePositionalEmbeddings(StatelessBlock):
+    def __init__(self,
+                 num_embed: int,
+                 dropout: float,
+                 max_seq_len: int,
+                 prefix: str) -> None:
+        self.num_embed = num_embed
+        self.dropout = dropout
+        self.max_seq_len = max_seq_len
+        self.prefix = prefix
+        self.embed_weight = mx.sym.Variable(prefix + "weight")
+
+    def __call__(self, data: mx.sym.Symbol,
+                 lengths: Optional[mx.sym.Symbol] = None,
+                 max_length: Optional[int] = None) -> mx.sym.Symbol:
+        assert max_length is not None, "max_length needed for positional embeddings."
```

Comment:
check_condition?

Reply:
didn't use check_condition as this isn't a user error. When pos embeddings are used max_length should also be not None.